packages/access-client/src/agent.js

```diff
@@ -327,18 +327,21 @@ export class Agent {
    * Creates a space signer and a delegation to the agent
    *
    * @param {string} name
+   * @param {object} [options]
+   * @param {'public'|'private'} [options.accessType] - The access type for the space. Defaults to 'public'.
    */
-  async createSpace(name) {
-    return await Space.generate({ name, agent: this })
+  async createSpace(name, { accessType = 'public' } = {}) {
+    return await Space.generate({ name, accessType, agent: this })
   }
 
   /**
    * @param {string} secret
    * @param {object} options
-   * @param {string} options.name
+   * @param {string} options.name - The name of the space.
+   * @param {'public'|'private'} [options.accessType] - The access type for the space. Defaults to 'public'.
    */
-  async recoverSpace(secret, { name }) {
-    return await Space.fromMnemonic(secret, { name, agent: this })
+  async recoverSpace(secret, { name, accessType = 'public' }) {
+    return await Space.fromMnemonic(secret, { name, accessType, agent: this })
```

Comment:
This kinda sucks for recovery - you have to remember the access type...

Reply:
Yeah. I'm not sure how to handle that. I will add a note in the code to come up with something better in the next iteration.